### DISCUSSION TITLE
Chain command can't download script from TFTP server by default

### DISCUSSION THREAD

**Discussion by ccy** [ipxe/ipxe#424] [General]
I embed the following script in release 1.0.0 of `ipxe.efi`:

```
#!ipxe
dhcp
chain main.ipxe
```
The `chain` command load `main.ipxe` from `next-server` via `TFTP` automatically.

After I upgrade to recent 1.21 `master` branch, the same `chain main.ipxe` prompt this error:

    Could not start download: Operation not supported (http://ipxe.org/3c092083D)

Is there any changes to the `chain` command?

**Comment by NiKiZe** [comment]
Can you verify that error message, url not working. (does it end up at https://ipxe.org/err/3c0920 ?)
What does ifstat show after this?

Please see the suggestions at that url, if that fails, build with DEBUG=open
Have you done any modifications to config/ files? such as removing TFTP support?

**Reply by ccy** [reply]
The build using untouch iPXE source code.  No changes to the `config/` too.

The TFTP works by using `chain tftp://${next-server}/main.ipxe`

I build the `ipxe.efi` with `DEBUG=open`.  Here is the outcome:

```
INTF 0x22b526a8+30 attempted to open unsupported URI scheme "<NULL>"
Could not start download: Operation not supported (http://ipxe.org/3c092083)
iPXE> ifstat
net0: 12:34:56:78:90:ab using i219v-10 on 0000:00:1f.6 (Ethernet) [open]
  [Link: up, TX:12 TXE:2 RX:3925 RXE:3644]
  [TXE: 1 x "Network unreachable (http://ipxe.org/28086090)"]
  [TXE: 1 x "Operation not supported (http://ipxe.org/3c654083)"]
  [RXE: 4 x "Invalid argument (http://ipxe.org/1c056082)"]
  [RXE: 3456 x "The socket is not connected (http://ipxe.org/380f6093)"]
  [RXE: 33 x "Operation not supported (http://ipxe.org/3c046083)"]
  [RXE: 148 x "Operation not supported (http://ipxe.org/3c046083)"]
```

**Reply by NiKiZe** [reply]
After reading it again, could you try with full `chain tftp://${next-server}/main.ipxe`
It might be that you don't have context.

If you are sure this worked before, can you do a [git bisect](https://ipxe.org/howto/bisect) to figure out where this possible bug was introduced?

**Reply by ccy** [reply]
Yes. `chain tftp://${next-server}/main.ipxe` works.  What is `context`?
